test/where-filter.test.js

```diff
@@ -29,17 +29,31 @@ context('whereFilter', () => {
 		assert.equal(array.filter(whereFilter({symbol: {neq: 'IBM'}})).length, 3);
 	});
 
-	it('should handle simple named relational operators', () => {
-		assert.equal(array.filter(whereFilter({qty: {eq: 10}})).length, 2);
-		assert.equal(array.filter(whereFilter({qty: {eq: 60}})).length, 1);
-
-		assert.equal(array.filter(whereFilter({qty: {neq: 10}})).length, 3);
-		assert.equal(array.filter(whereFilter({qty: {neq: 60}})).length, 4);
-
+	it('should handle other simple named relational operators', () => {
+		// lt/lte/gt/gte
 		assert.equal(array.filter(whereFilter({qty: {lt: 60}})).length, 2);
 		assert.equal(array.filter(whereFilter({qty: {lte: 60}})).length, 3);
 		assert.equal(array.filter(whereFilter({qty: {gt: 60}})).length, 1);
 		assert.equal(array.filter(whereFilter({qty: {gte: 60}})).length, 2);
+
+		// between
+		assert.equal(array.filter(whereFilter({qty: {between: [10, 60]}})).length, 3);
+
+		// like/nlike/ilike/nilike
+		assert.equal(array.filter(whereFilter({symbol: {like: 'OOG'}})).length, 1);
+		assert.equal(array.filter(whereFilter({symbol: {like: 'oog'}})).length, 0);
+		assert.equal(array.filter(whereFilter({symbol: {nlike: 'OOG'}})).length, 4);
+		assert.equal(array.filter(whereFilter({symbol: {nlike: 'oog'}})).length, 5);
+		assert.equal(array.filter(whereFilter({symbol: {ilike: 'OoG'}})).length, 1);
+		assert.equal(array.filter(whereFilter({symbol: {nilike: 'oOg'}})).length, 4);
+
+		// like with regex (it's just looking for any character that matches)
```

Comment:
I get the point re: taking the Regex tests out so as to avoid promoting that feature. But I don't follow re: SQL syntax, as far as I can tell the underlying implementation doesn't touch it at all.

Reply:
the `like` family of operators are expected to match `%` to any substring and `_` to a single character, and the existing implementation does convert `%` to `.*` and `_` to `.`.

But the code converts always; it doesn't support matching a literal `%` or `_`, only hacks like `\\x25`.  Postgres and mysql match `"\%"` to a literal `%`.

Another point of divergence, postgres and msyql anchor `like` patterns at the start and end, but where-filter does not (ie sql `foo%` matches `foo` and `foobar` but not `barfoo` (must start with "foo"), and `%bar` matches `bar` and `foobar` but not `barfoo` (must end with "bar")).

Reply:
see e.g. https://github.com/6RiverSystems/where-filter/pull/29 for a tentative fix to regex handling